client/lib/localforage/index.js

```diff
@@ -15,7 +16,35 @@ export const config = {
 	]
 };
 
-export function getLocalForage() {
+// Promise that resolves when our localforage configuration has been applied
+const localForagePromise = new Promise( ( resolve ) => {
 	localforage.config( config );
-	return localforage;
+	resolve( localforage );
+} );
+
+// Wraps a function to run after waiting until a promise has resolved.
+// The promise should contain the original object for context.
+const wrapOriginalFunc = ( promise, original ) => {
+	return function( ...args ) {
+		return promise.then( ( context ) => original.apply( context, args ) );
+	};
 }
+
+// List of localforage methods that should wait for localForagePromise to resolve
+const wrapFuncs = [
+	'getItem', 'setItem', 'removeItem',
+	'length', 'key', 'keys', 'iterate', 'clear'
+];
+
+// Proxy each localforage method to ensure our configuration is initialized first
+// NOTE: This means every localForage method returns a promise
```

Comment:
I believe every method is async except for config, so no real change to the api.